xrateskit/src/main/java/io/horizontalsystems/xrateskit/entities/GlobalCoinMarket.kt

```diff
@@ -1,20 +1,102 @@
+//-----------------------------------------------------------
+@Entity(indices = [Index(value = ["currencyCode", "timePeriod"], unique = true)])
+data class GlobalCoinMarketPointInfo(
+    @PrimaryKey(autoGenerate = true)
+    val id: Long = 0,
+    val currencyCode: String,
+    val timestamp: Long,
+    val timePeriod: TimePeriod
+){
+    @Ignore
+    val points: MutableList<GlobalCoinMarketPoint> = mutableListOf()
+}
+
+//-----------------------------------------------------------
+@Entity(foreignKeys = [
+    ForeignKey(
+        entity = GlobalCoinMarketPointInfo::class,
+        parentColumns = ["id"],
+        childColumns = ["pointInfoId"],
+        onDelete = CASCADE)]
+)
+data class GlobalCoinMarketPoint(
+    @PrimaryKey(autoGenerate = true)
+    val id: Long = 0,
+    val timestamp: Long,
+    val volume24h: BigDecimal,
+    val marketCap: BigDecimal,
+    val btcDominance: BigDecimal = BigDecimal.ZERO,
+    val defiMarketCap: BigDecimal = BigDecimal.ZERO,
+    val defiTvl: BigDecimal = BigDecimal.ZERO,
+    var pointInfoId: Long = 0
+)
+
+//-----------------------------------------------------------
 data class GlobalCoinMarket(
     val currencyCode: String,
-    var volume24h: BigDecimal,
+    val volume24h: BigDecimal,
     val volume24hDiff24h: BigDecimal,
-    var marketCap: BigDecimal,
+    val marketCap: BigDecimal,
     val marketCapDiff24h: BigDecimal,
-    var btcDominance: BigDecimal = BigDecimal.ZERO,
-    var btcDominanceDiff24h: BigDecimal = BigDecimal.ZERO,
-    var defiMarketCap: BigDecimal = BigDecimal.ZERO,
-    var defiMarketCapDiff24h: BigDecimal = BigDecimal.ZERO,
-    var defiTvl: BigDecimal = BigDecimal.ZERO,
-    var defiTvlDiff24h: BigDecimal = BigDecimal.ZERO
-)
+    val btcDominance: BigDecimal = BigDecimal.ZERO,
+    val btcDominanceDiff24h: BigDecimal = BigDecimal.ZERO,
+    val defiMarketCap: BigDecimal = BigDecimal.ZERO,
+    val defiMarketCapDiff24h: BigDecimal = BigDecimal.ZERO,
+    val defiTvl: BigDecimal = BigDecimal.ZERO,
+    val defiTvlDiff24h: BigDecimal = BigDecimal.ZERO,
+    val globalCoinMarketPoints: List<GlobalCoinMarketPoint>
+){
+
+    companion object {
+        fun calculateData(currencyCode: String, globalMarketPoints: List<GlobalCoinMarketPoint>): GlobalCoinMarket {
+
+            var marketCap = BigDecimal.ZERO
+            var marketCapDiff = BigDecimal.ZERO
+            var defiMarketCap = BigDecimal.ZERO
+            var defiMarketCapDiff = BigDecimal.ZERO
+            var volume24h = BigDecimal.ZERO
+            var volume24hDiff = BigDecimal.ZERO
+            var btcDominance = BigDecimal.ZERO
+            var btcDominanceDiff = BigDecimal.ZERO
+            var tvl = BigDecimal.ZERO
+            var tvlDiff = BigDecimal.ZERO
+
+            if(globalMarketPoints.isNotEmpty()){
+                val startingPoint = globalMarketPoints.last()
+                val endingPoint = globalMarketPoints.first()
+
+                marketCap = endingPoint.marketCap
```

Comment:
Can setting the value and its diff value be extracted to method to avoid code duplication?